api/tests/src/db/models/factories.py

```diff
@@ -316,6 +316,8 @@ class Params:
 
         no_current_summary = factory.Trait(current_opportunity_summary=None)
 
+        no_attachments = factory.Trait(opportunity_attachments=None)
```

Comment:
Added this trait and associated seeder update to make sure no PK violations hit (attachment ID was causing a problem with duplicate generation).

Reply:
You can work around that by adding something like what we have in the opportunity factory to the attachment factory:
```py
    @classmethod
    def _setup_next_sequence(cls):
        if _db_session is not None:
            value = _db_session.query(
                func.max(opportunity_models.Opportunity.opportunity_id)
            ).scalar()
            if value is not None:
                return value + 1

        return 1
```
Effectively, tell the factory to start counting differently

Alternatively - if we don't set the primary key ID at all in the factory - the DB will automatically set it (Postgres integer primary keys are automatically given a sequence they draw from).

Reply:
Removed setting the PK in the factory. Seems to work.